liquidctl/driver/nzxt_smart_device.py

```diff
@@ -182,66 +272,188 @@ def get_status(self, **kwargs):
-        else:
-            selected_channels = { channel: self._speed_channels[channel] }
-        for cname, (cid, smin, smax) in selected_channels.items():
-            if duty < smin:
-                duty = smin
-            elif duty > smax:
-                duty = smax
-            LOGGER.info('setting %s duty to %i%%', cname, duty)
-            self._write([0x2, 0x4d, cid, 0, duty])
+    def get_status(self, **kwargs):
+        """Get a status report.
+
+        Returns a list of (key, value, unit) tuples.
+        """
+        status = []
+        num_valid_replies_recvd = 0
+        msg_1101_reply = False
+        msg_2103_reply = False
+        msg_6702_reply = False
+        msg_6704_reply = False
+        accessory_name_dict = {
+            0x04: "HUE 2 LED Strip",
+            0x08: "HUE 2 Cable Comb", 
+            0x0a: "HUE 2 Underglow 200mm",
+            0x0b: "AER RGB 2 120mm", 
+            0x0c: "AER RGB 2 140mm"
+        }
+        # Get configuration information from device
+        LOGGER.debug('Issuing command 0x10 0x01 to get firmware info')
+        self._write([0x10, 0x01, 0x00, 0x00, 0x00, 0x00])
+        LOGGER.debug('Issuing command 0x20 0x03 to get lighting info')
+        self._write([0x20, 0x03, 0x00, 0x00, 0x00, 0x00])
+        # After issuing the above 2 commands, we will get a series of replies that
+        # will include everything we want to extract and display to the user.
+        # It may take 10 or 12 reply messages before we get all of the expected replies.
+        for x in range(12):
+            if num_valid_replies_recvd == 4:
+                break
+            msg = self.device.read(_READ_LENGTH_V2)
+            LOGGER.debug('received %s', ' '.join(format(i, '02x') for i in msg))
+            if not msg_1101_reply and msg[0] == 0x11 and msg[1] == 0x01:
+                fw = '{}.{}.{}'.format(msg[0x11], msg[0x12], msg[0x13])
+                status.append(('Firmware version', fw, ''))
+                num_valid_replies_recvd += 1
+                msg_1101_reply = True
+                continue
+            if not msg_2103_reply and msg[0] == 0x21 and msg[1] == 0x03:
+                num_light_channels = msg[14]  # the 15th byte (index 14) is # of light channels
+                accessories_per_channel = 6   # each lighting channel supports up to 6 accessories
+                light_accessory_index = 15    # offset in msg of info about first light accessory
+                for light_channel in range(num_light_channels):
+                    for accessory_num in range(accessories_per_channel):
+                        accessory_id = msg[light_accessory_index]
+                        light_accessory_index += 1
+                        if accessory_id in accessory_name_dict:
+                            status.append(('LED {} accessory {}'.format(light_channel + 1, accessory_num + 1),
+                                   accessory_name_dict.get(accessory_id), ''))
```

Comment:
Use `'Unknown'` instead of `''` as the default value.

Reply:
That's the third element of 3-tuple:

1st element: Item type/prefix (e.g. `LED 1 accessory 1`)
2nd element: Item name (e.g. `HUE 2 LED Strip`)
3rd element: Units (e.g. `rpm`, but for lighting accessories, the 3rd element is null rather than `Unknown`)

Reply:
Right, sorry about that.  Use `accessory_name_dict.get(accessory_id, 'Unknown'), ''))` then.

Also (minor), I prefer to align `accessory...` with `'LED {} ...`.  This also follows PEP-8.

Reply:
Because of ...
```Python
if accessory_id in _ACCESSORY_NAMES:
```
... it means the dictionary lookup will always succeed. However, I can remove this line and add `Unknown` as suggested:
```Python
self._ACCESSORY_NAMES.get(accessory_id, 'Unknown'), ''))
```
I've aligned `accessory...` with `LED {} ...`

Reply:
You're right.  And yes, please do that change.

I think displaying 'Unknown' motivates users to report it.